pkg/middlewares/accesslog/logger.go

```diff
@@ -238,6 +228,25 @@ func (h *Handler) ServeHTTP(rw http.ResponseWriter, req *http.Request, next http
 		return
 	}
 
+	defer func(capt capture.Capture, rw http.ResponseWriter) {
+		if h.config.BufferingSize > 0 {
+			h.logHandlerChan <- handlerParams{
+				logDataTable: logDataTable,
+			}
+			return
+		}
+
+		logDataTable.DownstreamResponse = downstreamResponse{
+			headers: rw.Header().Clone(),
+		}
+
+		logDataTable.DownstreamResponse.status = capt.StatusCode()
+		logDataTable.DownstreamResponse.size = capt.ResponseSize()
+		logDataTable.Request.size = capt.RequestSize()
+
+		h.logTheRoundTrip(logDataTable)
+	}(capt, rw)
+
```

Comment:
Hello @hood,

Thanks for submitting this!

At a glance, I have a few comments:
the `ClientUsername` field should be handled in the defer func
the defer func should replace the code after the `next.ServeHTTP` call
if `h.config.BufferingSize > 0` the `logDataTable` should be updated as well

Could you please fix these? As well, could you please add unit tests?

Thanks again!

Reply:
Hello @hood,

Thanks for the rework! Nonetheless, it looks like there is still an indentation issue.

> Do you have any pointers or ideas on how I could proceed?

The gist is not the best way to proceed (IMHO), please don't hesitate to push the test code on the PR branch, it would be easier for the review.

Reply:
> it looks like there is still an indentation issue
Mb, broke my editor and erroneously indented with spaces instead of tabs. 🤦 

I’ve pushed the WIP test code as request, but in general I agree that it doesn’t look like the best way to test the scenario. I’m not familiar enough with the codebase to come up with something better, so let’s discuss how we can best test that behaviour.